sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1192,6 +1192,13 @@ object SQLConf {
       .createWithDefault(
         "org.apache.spark.sql.execution.datasources.SQLHadoopMapReduceCommitProtocol")
 
+  val FILE_NAMING_PROTOCOL_CLASS =
+    buildConf("spark.sql.sources.namingProtocolClass")
```

Comment:
do we really need to allow users to customize the file naming?

Reply:
@cloud-fan - yes I think so. Before this PR, file naming is part of commit protocol, and each commit protocol checked in Spark code base, has their own specification for naming - `HadoopMapReduceCommitProtocol`, `PathOutputCommitProtocol` and `ManifestFileCommitProtocol`. So the external commit protocol we should expect they might already have their custom way of naming, and we should allow them to implement their own naming protocol.

Reply:
I agree that the file naming shouldn't be part of commit protocol, but I think the callers in Spark should decide the file name, like file format writer, or something similar for hive tables, not users.

Reply:
Updated per discussion offline. Keep naming protocol internal and not user-facing via config.